hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/TestDecimalTypeDataWorkflow.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi
+
+import org.apache.hudi.DataSourceWriteOptions._
+import org.apache.hudi.common.config.{HoodieReaderConfig, HoodieStorageConfig}
+import org.apache.hudi.config.HoodieWriteConfig
+import org.apache.hudi.testutils.SparkClientFunctionalTestHarness
+import org.apache.spark.sql.types.Decimal
+import org.apache.spark.sql.{DataFrame, SaveMode}
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+
+class TestDecimalTypeDataWorkflow extends SparkClientFunctionalTestHarness{
+  val sparkOpts: Map[String, String] = Map(
+    HoodieStorageConfig.LOGFILE_DATA_BLOCK_FORMAT.key -> "parquet",
+    HoodieWriteConfig.RECORD_MERGE_IMPL_CLASSES.key -> classOf[DefaultSparkRecordMerger].getName)
+  val fgReaderOpts: Map[String, String] = Map(
+    HoodieReaderConfig.FILE_GROUP_READER_ENABLED.key -> "true",
+    HoodieReaderConfig.MERGE_USE_RECORD_POSITIONS.key -> "true")
+  val opts = sparkOpts ++ fgReaderOpts
+
+  @Test
+  def testDecimalInsertUpdateDeleteRead(): Unit = {
```

Comment:
Can you add various precisions so we can test the different possible representations in parquet?
https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#decimal

Reply:
sg

Reply:
yeah, let's test at least 3 cases:

1. precision <= 10
2. precision > 10 && precision < 18
3. precision > 18

Reply:
I already did.

Reply:
`  @ParameterizedTest
  @CsvSource(value = Array("10,2", "20,10", "38,18", "5,0"))`
So, let me add one more (15, 10)

Reply:
Now:
`  @ParameterizedTest
  @CsvSource(value = Array("10,2", "15,5", "20,10", "38,18", "5,0"))`